### PR TITLE
Add limit to deal vote count manager

### PR DESCRIPTION
**Pull request by lisasgoh**
Split PR #130 - Deal Vote Count Manager limit on number of deals fetched

### CODE REVIEW

File: src/main/java/com/google/step/datamanager/DealVoteCountManagerDatastore.java [hunk 1]
```diff
@@ -34,22 +36,30 @@ public int getVotes(long dealId) {
...
         dealIdResults.add((long) entity.getProperty("deal"));
       }
+      dealIdResults.addAll(dealIdsArrayList);
     }
-    // Add those that have not been voted on and not in datastore to the end
-    dealIdResults.addAll(dealIdsArrayList);
+    dealIdResults =
+        limit == -1
+            ? dealIdResults
+            : dealIdResults.stream().limit(limit).collect(Collectors.toList());
```
Comment (anuragshirolkar) [inline src/main/java/com/google/step/datamanager/DealVoteCountManagerDatastore.java]: Why is this necessary if the fetch options in the prepared query already specified the fetch limit?

**Review by anuragshirolkar — DISMISSED** [review]

File: src/main/java/com/google/step/datamanager/DealVoteCountManagerDatastore.java (hunk 1 above)
Comment (lisasgoh) [reply src/main/java/com/google/step/datamanager/DealVoteCountManagerDatastore.java]: cos the entities retrieved are only those deals that have been voted on, so there are some deals that might not have been voted yet and wont be in the datastore, which is equivalent to a votecount of 0 and will have to be added into the list later on. So i changed the fetching to fetch all at first, then find the deals that have not been voted on and add into the result list at the appropriate index (where the vote count is 0) and limit the list after

**Review by anuragshirolkar — APPROVED** [review]
